Build (cross-compiling) error with trash 1.0.0 for armv7-linux-androideabi platform

**Describe the bug**
When trying to build (cross-compile) Nushell for armv7-linux-androideabi platform, it gives some errors: `use of undeclared type or module platform`


Full error list:
```Rust
Compiling trash v1.0.0
error[E0433]: failed to resolve: use of undeclared type or module `platform`
  --> D:\rust\cargo\registry\src\github.com-1ecc6299db9ec823\trash-1.0.0\src\lib.rs:49:5
   |
49 |     platform::remove(path)
   |     ^^^^^^^^ use of undeclared type or module `platform`

error[E0433]: failed to resolve: use of undeclared type or module `platform`
  --> D:\rust\cargo\registry\src\github.com-1ecc6299db9ec823\trash-1.0.0\src\lib.rs:71:5
   |
71 |     platform::remove_all(paths)
   |     ^^^^^^^^ use of undeclared type or module `platform`

error[E0433]: failed to resolve: use of undeclared type or module `platform`
  --> D:\rust\cargo\registry\src\github.com-1ecc6299db9ec823\trash-1.0.0\src\lib.rs:76:5
   |
76 |     platform::is_implemented()
   |     ^^^^^^^^ use of undeclared type or module `platform`

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0433`.
error: could not compile `trash`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**To Reproduce**
1. clone the repo
2. add this to `config` file in `.cargo`:
```YAML
[target.armv7-linux-androideabi]
// check your own path to SDK
ar = "D:\\androidSDK\\ndk-bundle\\toolchains\\llvm\\prebuilt\\windows-x86_64\\bin\\llvm-ar.exe"
linker = "D:\\androidSDK\\ndk-bundle\\toolchains\\llvm\\prebuilt\\windows-x86_64\\bin\\armv7a-linux-androideabi21-clang++.cmd"

[build]
target = "armv7-linux-androideabi"
```
3. `cargo build --release`

**Expected behavior**
Build correctly executable Nushell file for armv7-linux-androideabi platform.

**Configuration (please complete the following information):**
 - Host OS: Windows 10
- Target Platform: armv7-linux-androideabi
 - Version: stable

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -62,7 +62,7 @@ nu-build = { version = "0.12.0", path = "./crates/nu-build" }
 test-bins = []
 
 default = ["sys", "ps", "textview", "inc", "str"]
-stable = ["default", "starship-prompt", "binaryview", "match", "tree", "average", "post", "fetch", "clipboard-cli"]
+stable = ["default", "starship-prompt", "binaryview", "match", "tree", "average", "post", "fetch", "clipboard-cli", "trash-support"]
 
 # Default
 textview = ["crossterm", "syntect", "url", "nu_plugin_textview"]
@@ -82,6 +82,7 @@ tree = ["nu_plugin_tree"]
 
 clipboard-cli = ["nu-cli/clipboard-cli"]
 starship-prompt = ["nu-cli/starship-prompt"]
+trash-support = ["nu-cli/trash-support"]
 
 [[bin]]
 name = "fail"

--- a/crates/nu-cli/Cargo.toml
+++ b/crates/nu-cli/Cargo.toml
@@ -79,11 +79,11 @@ term = "0.5.2"
 termcolor = "1.1.0"
 textwrap = {version = "0.11.0", features = ["term_size"]}
 toml = "0.5.6"
-trash = "1.0.0"
 typetag = "0.1.4"
 umask = "0.1"
 unicode-xid = "0.2.0"
 
+trash = { version = "1.0.0", optional = true }
 clipboard = { version = "0.5", optional = true }
 starship = { version = "0.39.0", optional = true }
 
@@ -101,3 +101,4 @@ nu-build = { version = "0.12.0", path = "../nu-build" }
 stable = []
 starship-prompt = ["starship"]
 clipboard-cli = ["clipboard"]
+trash-support = ["trash"]

--- a/crates/nu-cli/src/commands/rm.rs
+++ b/crates/nu-cli/src/commands/rm.rs
@@ -12,6 +12,7 @@ pub struct Remove;
 pub struct RemoveArgs {
     pub rest: Vec<Tagged<PathBuf>>,
     pub recursive: Tagged<bool>,
+    #[allow(unused)]
     pub trash: Tagged<bool>,
 }
 

--- a/crates/nu-cli/src/shell/filesystem_shell.rs
+++ b/crates/nu-cli/src/shell/filesystem_shell.rs
@@ -946,7 +946,7 @@ impl Shell for FilesystemShell {
         RemoveArgs {
             rest: targets,
             recursive,
-            trash,
+            trash: _trash,
         }: RemoveArgs,
         name: Tag,
         path: &str,
@@ -1024,16 +1024,30 @@ impl Shell for FilesystemShell {
 
                 if let Ok(metadata) = f.symlink_metadata() {
                     if metadata.is_file() || metadata.file_type().is_symlink() || recursive.item || is_empty() {
-                        let result = if trash.item {
-                            trash::remove(f)
-                                .map_err(|e| f.to_string_lossy())
-                        } else if metadata.is_file() {
-                            std::fs::remove_file(f)
-                                .map_err(|e| f.to_string_lossy())
-                        } else {
-                            std::fs::remove_dir_all(f)
-                                .map_err(|e| f.to_string_lossy())
-                        };
+                        let result;
+                        #[cfg(feature = "trash-support")]
+                        {
+                            result = if _trash.item {
+                                trash::remove(f)
+                                   .map_err(|e| f.to_string_lossy())
+                            } else if metadata.is_file() {
+                                std::fs::remove_file(f)
+                                    .map_err(|e| f.to_string_lossy())
+                            } else {
+                                std::fs::remove_dir_all(f)
+                                    .map_err(|e| f.to_string_lossy())
+                            };
+                        }
+                        #[cfg(not(feature = "trash-support"))]
+                        {
+                            result = if metadata.is_file() {
+                                std::fs::remove_file(f)
+                                    .map_err(|e| f.to_string_lossy())
+                            } else {
+                                std::fs::remove_dir_all(f)
+                                    .map_err(|e| f.to_string_lossy())
+                            };
+                        }
 
                         if let Err(e) = result {
                             let msg = format!("Could not delete {:}", e);

